src/osd/ReplicatedBackend.cc

```diff
@@ -1981,14 +1981,14 @@ int ReplicatedBackend::build_push_op(const ObjectRecoveryInfo &recovery_info,
   if (available > 0) {
     if (!recovery_info.copy_subset.empty()) {
       interval_set<uint64_t> copy_subset = recovery_info.copy_subset;
-      bufferlist bl;
+      interval_set<uint64_t> fiemap_included;
+      map<uint64_t, uint64_t> m;
       int r = store->fiemap(ch, ghobject_t(recovery_info.soid), 0,
-                            copy_subset.range_end(), bl);
+                            copy_subset.range_end(), m);
       if (r >= 0)  {
-        interval_set<uint64_t> fiemap_included;
-        map<uint64_t, uint64_t> m;
-        bufferlist::iterator iter = bl.begin();
-        ::decode(m, iter);
+	// TODO: bluestore uses interval_set internally, right now
+	// we move data from interval_set to map, then back from map
+	// to interval_set...
```

Comment:
i'd just make an interval_set ctor that takes the stl map and swap()'s it into place